### DISCUSSION TITLE
How to query related data from 2 table?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [supabase/supabase#1386] [Questions] [answered]
Hi, I have table A there some columns
name
categories : array of ids in table B

table B
id
cate_name

I want to get data in table A but want get cate_name in table B too. I tried foreign key relation but it just use for connect 1-1 not 1-many
`await supabaseAdmin.from("A").select("*")... `

the final data I want is

name
categories:[ list of cate_name not the ids ]

**Comment by IRediTOTO** [comment]
Let it go :| I connect direct to database. Here example code. 
https://node-postgres.com/features/queries
```

import { Client } from "pg";

const client = new Client({
    user: 'postgres',
    host: 'db.321.supabase.co',
    database: 'postgres',
    password: '123',
    port: 5432,
})


export async function get(params) {
    client.connect()
    const data = await client.query(`select * from auth.users`).then(res => res);
    // console.log("data",data)



    return {
        body: {
           data
        }
    }
}
```

**Comment by steve-chavez** [comment, marked as answer]
> categories : array of ids in table B

Array foreign keys are not supported, see this previous discussion: https://github.com/supabase/supabase/discussions/535#discussioncomment-285696.

If you define a regular join/junction table you can query through supabase-js like mentioned on the [docs](https://supabase.io/docs/reference/javascript/select#query-foreign-tables).
